### DISCUSSION TITLE
Data set templates

### DISCUSSION THREAD

**Discussion by katelynienaber** [zowe/zowe-explorer-vscode#1133] [Ideas]
Hey guys! I am opening this discussion because I am working on it. I will add some images/gifs as the programming comes along.

How I imagine this could work:
- Templates exist in `settings.json` (or in the giant `config.json` file that is being developed)
- New templates can be added/edited/deleted from the Profiles menu
- While you are allocating a data set, you can create a template too.
Steps: Enter all your parameters
Select `Create template from this configuration`
A VSCode alert appears that says, "Template created"...but the Allocate dialog will not disappear, you can still change parameters and such
Now you can select `Create data set` from the list...or you can make another template...or whatever.

**Comment by katelynienaber** [comment]
Current branches related to this:
- Validation during DS allocation: `ds-attribute-validation`
- Allocate DS from JSON file: `allocate-from-json`
- Create template from JSON file: `template-from-json`
- Edit/delete templates: `delete-templates`
- Create template from allocate dialog: `template-from-new-ds-dialog`
- Create template from scratch (button in context menu): `new-template-menu-item`

**Comment by katelynienaber** [comment]
Linked CLI issue: https://github.com/zowe/zowe-cli/issues/947
Linked VSCode issue for adding a divider to a QuickPick list: https://github.com/microsoft/vscode/issues/74967

Ideas:
- Allow people to use variables in the JSON files
- Allow organizations/people to change many templates at one go (not one by one)
- Talk to @MikeBauerCA about implementing this for some other projects...should talk again to understand the use case

**Comment by jellypuno** [comment]
I'm excited to see this 😄
